src/models/user.js

```
@@ -8,10 +8,10 @@ const userSchema = new mongoose.Schema({
   password: {}
 });
 
-userSchema.statics.findByLogin = async function(login) {
-  const user = await this.findOne({
-    email: login
-  });
+userSchema.statics.findByEmail = async function(email) {
+  const user = await this.findOne({ email: email })
+    .exec()
+    .catch(e => null);
```

why do you hide error? This will lead to unforeseen consequences

---

идея была в том, чтобы функция всегда возвращала (промис) со значением, если все ок, и null, если что-то не так (чтоб промис всегда резолвился в конце, а не выкидывал потом исключение при await). наверное правильнее делать вот так: https://stackoverflow.com/questions/50210289/get-data-using-await-async-without-try-catch
но я тогда этого варианта не знала. мне переделать?

---

Подход с `try/catch` лучше на мой взгляд. Выглядит лучше и код читабельней.

---

https://blog.grossman.io/how-to-write-async-await-without-try-catch-blocks-in-javascript/

---

Как раз статья эта тому подтверждение)

---

странно, вроде ж статья называется,к ак не использовать try/catch и сам автор пишет:
"In async/await functions it is common to use try/catch blocks to catch such errors.

I'm not coming from a typed language background, so the try/catch adds for me additional code that in my opinion doesnt look that clean. I'm sure it's a matter of personal preference, but that's my opinion."

---

У автора чисто субъективное мнение, как он пишет in my opinion) Дело вкуса. Я высказал просто тоже свое мнение )

---

И метод `exec()` я не понял для чего там.

---

Проблема в том что ты прячешь эрроры, все. Проблема не в трай кэтчь или промисах

---

exec нужен, чтобы из query получить настоящий промис, так написано в документации :
"Mongoose queries are not promises. They have a .then() function for co and async/await as a convenience. If you need a fully-fledged promise, use the .exec() function."
https://mongoosejs.com/docs/promises.html